sympy/parsing/sympy_parser.py

```diff
@@ -905,8 +908,21 @@ def eval_expr(code, local_dict, global_dict):
     return expr
 
 
+_refloat = regex.compile(r"(((\d*\.\d+)|(\d+\.?\d?))([eE][-+]?\d+)?)")
```

Comment:
Maybe import this from the `tokenize` module (https://github.com/python/cpython/blob/44a2c4aaf2d0c03c70646eb16fbc6c1ba1689e69/Lib/tokenize.py#L62). It isn't part of the public API, but it doesn't seem to have changed at all since 2.7.

Reply:
> Maybe import this from the `tokenize` module

The expressions have changed between 2.7 and 3.6. I believe this is where my pattern came from, however. In addition, the Intnumber includes alternate bases (octal, ...) that are not of concern at the preproccessing stage.

Reply:
> The expressions have changed between 2.7 and 3.6. 

I see you removed this in an edit, but I just wanted to point out that presently, parse_expr uses the standard library tokenize, meaning it only supports language features from the version of Python it is running in. 

```py
# python 3.6
>>> from sympy.parsing.sympy_parser import parse_expr
>>> parse_expr('1_2')
12
```

```py
# python 2.7
>>> from sympy.parsing.sympy_parser import parse_expr
>>> parse_expr('1_2')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "sympy/parsing/sympy_parser.py", line 966, in parse_expr
    return eval_expr(code, local_dict, global_dict)
  File "sympy/parsing/sympy_parser.py", line 879, in eval_expr
    code, global_dict, local_dict)  # take local objects in preference
  File "<string>", line 1
    Integer (1 )Symbol ('_2' )
                     ^
SyntaxError: invalid syntax
```

I think it's fine to keep things this way even for the extended syntax. Backporting language features is something that's doable, but not really worth the added complexity IMO.